materialize-sql/driver.go

```diff
@@ -141,15 +146,34 @@ func (d *Driver) ApplyUpsert(ctx context.Context, req *pm.ApplyRequest) (*pm.App
 				loadedBinding.FieldSelection.String(),
 			)
 		} else if loadedBinding != nil {
-			continue // Table was already created. Nothing to do.
+			// If the table has already been created, make sure all non-null fields
+			// are marked as such
+			for _, field := range bindingSpec.FieldSelection.Values {
+				var p = bindingSpec.Collection.GetProjection(field)
+
+				if p.Inference.Exists != pf.Inference_MUST {
```

Comment:
If I'm reading this correctly, any time a proposed spec contains fields that are not required the alter table statement will be executed. Would it be possible to compare the proposed spec vs. the loaded spec and only run the table alteration if there's a difference?

Reply:
done ✅